python/test_infra/test_utils.py

```diff
@@ -94,7 +95,15 @@ def decorator(test_fn: Callable):
         # test_fn should be called with no args.
         def wrapper():
             ctx = Context()
-            loc = Location.unknown(ctx)
+
+            # Grab the location of the test function in python for later debugging
+            try:
+                fname = inspect.getfile(test_fn)
+                line_no = inspect.getsourcelines(test_fn)[1]
+                loc = Location.file(fname, line_no, 0, ctx)
```

Comment:
Is there a way to get column information?  I'm just thinking we might need a way to disambiguate compound expressions like:
```python
out = (a + b) * c
```

Reply:
I will look into it, but I don't believe so / think it matters in this case, just because this annotates the generated function with the decorated function location, not the actual operands within the function. If we want to get that fine grained though I can look into that as well

Reply:
Oh I see, yeah we want to be more fine grained than this.  We want the location of the statement inside the test file, so we need to use `inspect` from inside the builder, i.e. if we had test:
```python
200 def test_arbitrary_op_chain_ttnn(
201     in0: Operand, in1: Operand, in2: Operand, builder: TTIRBuilder
202 ):
203     add = builder.add(in0, in1)
204     exp = builder.exp(in2)
205     return builder.multiply(add, exp)
```

We would expect the following locations to be populated:
- `ttir.add loc("python/test_infra/test_ttir_ops_ttnn.py:203")`
- `ttir.exp loc("python/test_infra/test_ttir_ops_ttnn.py:204")`
- `ttir.multiply loc("python/test_infra/test_ttir_ops_ttnn.py:204")`

Ideally we can get column info as well so we could support statements with compound expressions.  Every operation intermediate needs a unique location, so if we cannot get the column, we'll have to manually track a counter for each op on a single line to act as a unique id, disambiguating them.

Reply:
Ah ok makes sense, I also now see why `traceback` was your initial solution idea instead of `inspect`, since the locations will be precisely where the `TTIRBuilder` calls are made

Reply:
Yeah we might want to tag the wrapped test function in some way so that the traceback knows where to stop:

```python
def decorator(test_fn: Callable):
    test_fn._traceback_stop = True
```

Some way to uniquely tag the stack frame that we actually care about.

Reply:
Ah that's not a bad idea, but it's not very general if other consumers want to use the `TTIRBuilder`, i was thinking something along the lines of stopping at the first frame outside of `ttir_builder.py`

Reply:
Interesting idea, yeah that sounds worth a shot